src/main/java/bg/libapp/libraryapp/service/RentService.java

```diff
@@ -0,0 +1,193 @@
+
+import java.time.LocalDate;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static bg.libapp.libraryapp.model.constants.ApplicationConstants.ONE_MONTH;
+
+@Transactional
+@Service
+public class RentService {
+    private final Logger logger = LoggerFactory.getLogger(BookService.class);
+    private final BookRepository bookRepository;
+
+    private final RentRepository rentRepository;
+
+    private final UserService userService;
+
+    private final BookService bookService;
+
+    @Autowired
+    public RentService(BookRepository bookRepository, RentRepository rentRepository, UserService userService, BookService bookService) {
+        this.bookRepository = bookRepository;
+        this.rentRepository = rentRepository;
+        this.userService = userService;
+        this.bookService = bookService;
+    }
+
+    public RentDTO rentBook(RentAddRequest rentAddRequest, String isbn) {
+        Book book = bookService.findBookByIsbn(isbn);
+        hasEnoughAvailableQuantity(book);
+
+        User borrower = userEligibleToRent(rentAddRequest);
+        logger.info("rentBook method called with borrower username '" + borrower.getUsername()
+                + "' for book with isbn '" + isbn
+                + "' with requestParams: " + rentAddRequest);
+
+        checkRequirementsForRentCreation(isbn, borrower);
+        Rent rent = new Rent()
+                .setRentDate(LocalDate.now())
+                .setBook(book)
+                .setUser(borrower)
+                .setExpectedReturnDate(getExpectedDateToSet(rentAddRequest));
+        rentRepository.saveAndFlush(rent);
+        bookRepository.saveAndFlush(book.setAvailableQuantity(book.getAvailableQuantity() - 1));
+        logger.info("Created new rent with params: " + rent);
+        return RentMapper.mapToRentDTO(rent);
+    }
+
+    private void hasEnoughAvailableQuantity(Book book) {
+        logger.info("hasEnoughAvailableQuantity method called for book with isbn '" + book.getIsbn() + "'.");
+        int availableQuantity = book.getAvailableQuantity();
+        if (availableQuantity < 1) {
+            logger.error("Insufficient available quantity for book with isbn '" + book.getIsbn()
+                    + "', because available quantity is =" + availableQuantity);
+            throw new InsufficientAvailableQuantityException(book.getIsbn(), availableQuantity);
+        }
+    }
+
+    private User userEligibleToRent(RentAddRequest rentAddRequest) {
+        String username = SecurityContextHolder.getContext().getAuthentication().getName();
+        logger.info("userEligibleToRent method called with params: " + rentAddRequest + "\n" +
+                " by authenticated user with username '" + username + "'.");
+        User borrower = userService.getUserByUsername(username);
+        String userRole = Role.values()[borrower.getRole()].toString();
+        Long userIdToRent = rentAddRequest.getUserId();
+        if (!userRole.equals("ADMIN")
```

Comment:
Use the enum values for the roles (or at least extract the string values in constants)